### DISCUSSION TITLE
How do you (users) deal with the adblocker being mediocre?

### DISCUSSION THREAD

**Discussion by peternrdstrm** [bromite/bromite#1484] [Q&A]
I've seen many poeople switch to Bromite Extended, and Energized Blu, but does anyone have a solution, that adresses 
A. CSS rules
B. More ads blocked

**Comment by HappyBullfrog** [comment]
I consider AdGuard the single most important piece of software on my cell phone and use it with the default AdGuard DNS server which can be accessed by any software or browser, even when not using the AdGuard software. 

This AdGuard DNS server also blocks the most egregious malware-serving sites.

The default AdGuard DNS server does not maintain any logs and seems to work well, even with the new DNS-over-QUIC protocol that they initially considered experimental.

There is a long list of other DNS servers you can choose from, but I no longer find any reason to look at them.

AdGuard blocks most ads in most software and usually does not cause any problems. Occasionally I have to toggle it off when there is a problem with a specific website. However that website can then be listed in AdGuard as an exception. I only have about a dozen exceptions.

One caution however. AdGuard does NOT work with any VPN other than it's own "AdGuard VPN" still under development and a little buggy. Specifically, some of the foreign and remote VPN servers are slow and not all can be accessed. But if you only want to disguise your location, the New York area server works well, probably because that server must handle a huge volume of financial transactions.

AdGuard is commercial software, but it is affordable.

**Reply by csagan5** [reply]
AdGuard does not solve the problem of lack of cosmetic filtering and most of the ads that can be blocked via DNS are already blocked with the default Bromite filters.

**Reply by HappyBullfrog** [reply]
I was only trying to report that the combination of the two works quite well. 

I disabled Bromite's ad blocking yesterday after I read other users' complaints about that feature slowing Bromite down. So far I don't see any difference, but Bromite has always seemed quite peppy to me, so it has become my default browser.

However I also use lots of other software and am no longer plagued by ads and pop-ups from software downloaded from the Google Play Store. 

We all need to select a DNS server; AdGuard's DNS servers are free to anyone using any software and maintain no logs, so seem worthy of consideration. They offer a second server variation which does no filtering, and a third which filters out content that might be objectionable for children.

However the AdGuard software itself offers a variety of features for filtering specific kinds of content, such as ads on foreign websites. They update those filters continuously, a few times per day. That content filtering seems to be beyond the privacy objectives of Bromite's capable programmers.

I am not a programmer or website designer so cannot consider CSS filtering. Adguard does provide options for blocking custom fonts, but recommends against doing so, because it will break some websites.

**Reply by Disqu3-mirOir-qt** [reply]
> AdGuard does not solve the problem of lack of cosmetic filtering and most of the ads that can be blocked via DNS are already blocked with the default Bromite filters.

AdGuard solved , have cosmectic and scriplet support.

![Screenshot_20220109-113209.jpg](https://user-images.githubusercontent.com/41585567/148678578-29413709-c1c3-4203-b162-eea3b19ef7ee.jpg)


![Screenshot_20220109-113127.jpg](https://user-images.githubusercontent.com/41585567/148678589-0e47f246-0e20-46a9-94eb-ce92fce83c89.jpg)

![Screenshot_20220109-113105.jpg](https://user-images.githubusercontent.com/41585567/148678594-17af2dab-df13-4421-80d0-dbee985e9ef9.jpg)

![Screenshot_20220109-113043.jpg](https://user-images.githubusercontent.com/41585567/148678598-94643d11-fec2-41e9-8961-8e71c5170006.jpg)

**Reply by HappyBullfrog** [reply]
I am in awe of people who can explain this so clearly that even I can understand it. Thank you!
